src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/SdkClient/ResourceManagerSdkClient.cs

```diff
@@ -707,7 +707,7 @@ public virtual PSResourceGroupDeployment ExecuteDeployment(PSCreateResourceGroup
                 WriteVerbose(ProjectResources.TemplateValid);
             }
 
-            ResourceManagementClient.Deployments.CreateOrUpdateAsync(parameters.ResourceGroupName, parameters.DeploymentName, deployment);
+            ResourceManagementClient.Deployments.CreateOrUpdate(parameters.ResourceGroupName, parameters.DeploymentName, deployment);
```

Comment:
We have a pending PR to change this to BeginCreateOrUpdate instead, here: https://github.com/Azure/azure-powershell/pull/2451/files#diff-0e1fbafcda48a1d6c2fefd0fec655bfcR719

CreateOrUpdate is a synchronous call, which blocks the thread execution. This will mean you won't get the progress messages in the console when the deployment is running.